react/features/toolbar/actions.web.js

```diff
@@ -0,0 +1,307 @@
+
+            toolbarTimeoutObject = setTimeout(() => {
+                dispatch(hideToolbar());
+            }, toolbarTimeout);
+        } else {
+            dispatch(setToolbarVisibility(false));
+            dispatch(setSubjectSlideIn(false));
+        }
+    };
+}
+
+/**
+ * Action that reset always visible toolbar to default state.
+ *
+ * @returns {Function}
+ */
+export function resetAlwaysVisibleToolbar(): Function {
+    return (dispatch: Dispatch<*>) => {
+        const alwaysVisible = config.alwaysVisibleToolbar === true;
+
+        dispatch(setAlwaysVisibleToolbar(alwaysVisible));
+    };
+}
+
+/**
+ * Signals that unclickable property of profile button should change its value.
+ *
+ * @param {boolean} unclickable - Shows whether button is unclickable.
+ * @returns {Function}
+ */
+export function setProfileButtonUnclickable(unclickable: boolean): Function {
+    return (dispatch: Dispatch<*>) => {
+        const buttonId = 'toolbar_button_profile';
+
+        dispatch(setToolbarButton(buttonId, {
+            unclickable
+        }));
+
+        UIUtil.removeTooltip(
+            document.getElementById('toolbar_button_profile'));
+    };
+}
+
+/**
+ * Shows desktop sharing button.
+ *
+ * @returns {Function}
+ */
+export function showDesktopSharingButton(): Function {
+    return (dispatch: Dispatch<*>) => {
+        const buttonId = 'toolbar_button_desktopsharing';
+        const isVisible = APP.conference.isDesktopSharingEnabled
+            && UIUtil.isButtonEnabled('desktop');
+
+        dispatch(setToolbarButton(buttonId, {
+            hidden: !isVisible
+        }));
+    };
+}
+
+/**
+ * Shows or hides the dialpad button.
+ *
+ * @param {boolean} show - Flag showing whether to show button or not.
+ * @returns {Function}
+ */
+export function showDialPadButton(show: boolean): Function {
+    return (dispatch: Dispatch<*>) => {
+        const shouldShow = UIUtil.isButtonEnabled('dialpad') && show;
+        const buttonId = 'toolbar_button_dialpad';
+
+        if (shouldShow) {
+            dispatch(setToolbarButton(buttonId, {
+                hidden: false
+            }));
+        }
+    };
+}
+
+/**
+ * Shows recording button.
+ *
+ * @returns {Function}
+ */
+export function showRecordingButton(): Function {
+    return (dispatch: Dispatch<*>) => {
+        const eventEmitter = APP.UI.eventEmitter;
+        const buttonId = 'toolbar_button_record';
+
+        dispatch(setToolbarButton(buttonId, {
+            hidden: false
+        }));
+
+        Recording.init(eventEmitter, config.recordingType);
+    };
+}
+
+/**
+ * Shows or hides the 'shared video' button.
+ *
+ * @returns {Function}
+ */
+export function showSharedVideoButton(): Function {
+    return (dispatch: Dispatch<*>) => {
+        const buttonId = 'toolbar_button_sharedvideo';
+        const shouldShow
+            = UIUtil.isButtonEnabled('sharedvideo')
+                && !config.disableThirdPartyRequests;
+
+        if (shouldShow) {
```

Comment:
Yep, in this case !shouldShow will always be equal to `false` but let's replace literal by expression.